torchqdynamics/utils/td_tensor.py

```diff
@@ -0,0 +1,165 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import get_args
+
+import torch
+from torch import Tensor
+
+from .tensor_types import OperatorLike, TDOperatorLike, cdtype, rdtype, to_tensor
+
+
+def to_tdtensor(
+    x: TDOperatorLike,
+    dtype: torch.dtype | None = None,
+    device: torch.device | None = None,
+    is_complex: bool = False,
+) -> TDTensor:
+    """Convert a `TDOperatorLike` object to a `TDTensor` object."""
+    if is_complex:
+        dtype = cdtype(dtype)
+
+    if isinstance(x, get_args(OperatorLike)):
+        x = to_tensor(x, dtype=dtype, device=device, is_complex=is_complex)
+        return ConstantTDTensor(x, dtype=dtype, device=device)
+    elif callable(x):
+        if dtype is None and not is_complex:
+            dtype = rdtype(dtype)
+        if device is None:
+            device = torch.device('cpu')
```

Comment:
```suggestion
```
I don't think we need this?

Reply:
Sorry we do now that I read the whole code, but we should catch the default device rather than `'cpu'`. And the first condition can be simplified to `if dtype is None`.